httpclient/src/main/java/org/apache/http/impl/execchain/MainClientExec.java

```diff
@@ -404,7 +404,7 @@ void establishRoute(
                         timeout > 0 ? timeout : 0,
                         context);
                 final HttpHost proxy  = route.getProxyHost();
-                tracker.connectProxy(proxy, false);
+                tracker.connectProxy(proxy, route.isSecure());
```

Comment:
@gknapowski it should most likely be `route.isSecure() && !route.isTunnelled()`, that is a direct (non-tunneled) connection to the proxy can be considered secure if the route is considered secure.

Reply:
I agree.  I tested with use cases.  I also tested with a secure connection to the Proxy with Tunneling.  Both work.

Reply:
@gknapowski Could you please squash the commits into a single one?

Reply:
Sorry about that.